inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/model/ResolvedGenericDataProviderConfig.java

```diff
@@ -0,0 +1,82 @@
+package rocks.inspectit.oce.core.instrumentation.config.model;
+
+import lombok.Builder;
+import lombok.Value;
+import org.apache.commons.lang3.tuple.Pair;
+import rocks.inspectit.oce.bootstrap.instrumentation.IGenericDataProvider;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Contains all properties necessary for defining a {@link IGenericDataProvider}.
+ * Instances of this class are derived by the {@link rocks.inspectit.oce.core.instrumentation.config.InstrumentationConfigurationResolver} from
+ * corresponding {@link rocks.inspectit.oce.core.config.model.instrumentation.GenericDataProviderConfig} objects.
+ */
+@Value
+@Builder
+public class ResolvedGenericDataProviderConfig {
+
+    /**
+     * The name uniquely identifying this provider.
+     */
+    private String name;
+
+    /**
+     * A map defining which call arguments of the instrumented method the provider requires and what he expects as their type.
+     * These are usually defined using the input paramters anmed "arg0","arg1" etc
+     */
+    @Builder.Default
+    private HashMap<Integer, String> expectedArgumentTypes = new HashMap<>();
+
+    /**
+     * The type of the return value of the instrumented method the data provider expects.
+     * If the data provider does not use the return value, this field is null.
+     */
+    private String expectedReturnValueType;
+
+    /**
+     * The type of the "this"-object of the instrumented method the data provider expects.
+     * If the data provider does not use the this-instance, this field is null.
+     */
+    private String expectedThisType;
+
+    /**
+     * The type of the thrown exception the data provider expects.
+     * If the data provider does not make use of the thrown exception, this field is null.
+     * <p>
+     * In contrast to the other fields, this type implies control flow.
+     * If a throwable is thrown, which is not a subtype of this type, the provider automatically return null.
+     */
+    private String expectedThrowableType;
```

Comment:
do we really need this? I mean can we just say it's throwable and that's it.. I mean you could say that this is needed in some crazy cases, but in general.. not sure, let's talk about it..